pdf/core/parser.go

```diff
@@ -708,6 +708,13 @@ func (parser *PdfParser) parsePdfVersion() (int, int, error) {
 			return 0, 0, err
 		}
 
+		// Create a new offset reader that ignores the invalid data before
+		// the PDF version.
+		parser.rs, err = newOffsetReader(parser.rs, parser.GetFileOffset()-8)
+		if err != nil {
+			return 0, 0, err
+		}
+
```

Comment:
I will create test cases for the offset of the reader after parsing the version. 
The  seekPdfVersionTopDown offset will be right on the % (of the %PDF1.X version), because the offset reader does set the file offset at that position. However, the straight away version match will have the offset right after the first 20 bytes. The pull request did not modify this behavior. It does not really matter because the parsing of xrefs takes the buffer pointer at end of the ReadSeeker right away. Should I reset the file pointer to the 0 position just as a precaution?

Reply:
Added version parsing test case.